Packs/CommonWidgets/Scripts/RSSWidget/RSSWidget.py

```diff
@@ -0,0 +1,124 @@
+import email.utils
+from datetime import datetime
+from time import mktime
+
+import feedparser
+from feedparser.util import FeedParserDict
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+
+class Client(BaseClient):
+    """Client for RSS Feed.
+    Attributes:
+        server_url(str): The RSS URL.
+        use_ssl: Whether to use ssl.
+        proxy(str): Use system proxy.
+    """
+
+    def __init__(self, server_url, use_ssl, proxy):
+        super().__init__(base_url=server_url, proxy=proxy, verify=use_ssl)
+
+    def get_feed_data(self):
+        """Retrieves the data from the RSS feed.
+        """
+        return self._http_request(method='GET', resp_type='response')
+
+
+def parse_feed_data(feed_response: requests.Response) -> FeedParserDict:
+    """Parses the data from the RSS feed.
+
+    Args:
+        feed_response (requests.Response): The raw data from the RSS feed.
+
+    Returns:
+        FeedParserDict: Parsed RSS feed data.
+    """
+    try:
+        return feedparser.parse(feed_response.text)
+    except Exception as err:
+        raise DemistoException(f"Failed to parse feed.\nError:\n{str(err)}")
+
+
+def collect_entries_data_from_response(parsed_feed_data: FeedParserDict) -> List[Dict[str, Any]]:
+    """Collects relevant data from the parsed RSS feed entries.
+
+    Args:
+        parsed_feed_data (FeedParserDict): Parsed RSS feed data.
+
+    Returns:
+        List[Dict[str, Any]]: The data from the RSS feed relevant for the widget.
+    """
+    entries_data: List[Dict[str, Any]] = []
+    if not parsed_feed_data:
+        raise DemistoException("Could not parse feed data.")
+
+    for entry in reversed(parsed_feed_data.entries):
+        if entry:
+            published = email.utils.parsedate(entry.published)
+            if not published:
+                continue
+
+            published_dt = datetime.fromtimestamp(mktime(published))
+            published_formatted = published_dt.strftime('%B %-d, %Y %-I:%M %p')
+
+            entries_data.append(
+                {
+                    'timestamp': published_formatted,
+                    'link': entry.get('link'),
+                    'title': entry.get('title'),
+                    'summary': entry.get('summary')
+                }
+            )
+
+    return entries_data
+
+
+def create_widget_content(entries_data: List[Dict[str, Any]]) -> str:
+    """Creates the human readable text for the widget.
+
+    Args:
+        entries_data (List[Dict[str, Any]]): The data from the  RSS feed relevant for the widget.
+
+    Returns:
+        str: The widget's content.
+    """
+    content: str = ''
+
+    for entry_data in entries_data:
+        content += f'## [{entry_data["title"]}]({entry_data["link"]})\n'
+        content += f'_{entry_data["timestamp"]}_\n'
+        content += f'#### {entry_data["summary"]}\n'
+        content += '---\n'
```

Comment:
### pure suggestion
you can add the template as a constant to make it more readable:
```
TEMPLATE = '''## [{title}]({link})
_{timestamp}_
#### {summary}
---
'''
```
and here:
```suggestion
    for entry_data in entries_data:
        content += TEMPLATE.format(**entry_data)
```

what do you think? not sure which I like better